core/src/lib.rs

```diff
@@ -6,6 +6,7 @@
 #![deny(non_snake_case)]
 #![deny(unused_mut)]
 #![deny(missing_docs)]
+#![feature(result_map_or_else)]
```

Comment:
We should avoid using unstable features, as they are only supported in nightly rust.